docs/book/v3/cookbook/double-pass-middleware.md

```diff
@@ -0,0 +1,181 @@
+# Using Double-Pass Middleware
+
+Expressive uses [PSR-15](https://www.php-fig.org/psr/psr-15/) middleware and
+request handlers exclusively as of version 3.
+
+In previous releases, however, we supported "double-pass" middleware, and a
+number of third-party packages provided double-pass middleware. How can you use
+this middleware with Expressive 3?
+
+> ### What is Double-Pass Middleware?
+>
+> Double pass middleware receives both the request and a response in addition to
+> the handler, and passes both the request and response to the handler when
+> invoking it:
+>
+> ```php
+> function (ServerRequestInterface $request, ResponseInterface $response, callable $next)
+> {
+>     $response = $next($request, $response);
+>     return $response->withHeader('X-Test', time());
+> }
+> ```
+>
+> It is termed "double pass" because you pass _both_ the request _and_ response when
+> delegating to the next layer.
+
+## doublePassMiddleware function
+
+zend-stratigility v2.2 and v3.0 ship a utility function,
+`Zend\Stratigility\doublePassMiddleware()`, that will decorate a callable
+double-pass middleware using a `Zend\Stratigility\Middleware\DoublePassMiddlewareDecorator` 
+instance; this latter is a PSR-15 impelementation, and can thus be used in your
+middleware pipelines.
+
+The function (and class) also expects a [PSR-7](https://www.php-fig.org/psr/psr-7/)
+`ResponseInterface` instance as a second argument; this is then passed as the
+`$response` argument to the double-pass middleware. The following examples
+demostrate both piping and routing to double pass middleware using this
+technique, and using zend-diactoros to provide the response instance.
+
+```php
+use Zend\Diactoros\Response;
+use function Zend\Stratigility\doublePassMiddleware;
```

Comment:
I would add one line before importing function to make it clearer, that we import function. Usually we do the same in our source, to have all import section separated be single empty line.

... and as I just noted it will be consistent with next example where we have this one empty line.

Reply:
Fixed.